src/port1.0/portutil.tcl

```diff
@@ -508,6 +508,17 @@ proc default_check {optionName index op} {
     }
 }
 
+# get_default_env
+# Unless port is Xcode dependent, returns env which sets developer_dir to
+# Command Line Tools
+proc get_default_env {} {
+    global use_xcode
+    if {$use_xcode eq "yes"} {
+        return ""
+    }
+    return "DEVELOPER_DIR=/Library/Developer/CommandLineTools"
```

Comment:
I doubt this works with all the Xcode and OS versions that base runs on.

Reply:
After reading up, I see that the other possible place for CommandLineTools is `"/"` for `$macosx_version < 10.9`
https://github.com/macports/macports-base/blob/1d6588d178b251c86abc37cbea5ab580e2b85fd1/src/port1.0/portutil.tcl#L3316
I tested it, and it looks like it works to use `DEVELOPER_DIR="/"` for both cases.
```
$ DEVELOPER_DIR="/" clang -v
Apple LLVM version 10.0.0 (clang-1000.10.44.4)
Target: x86_64-apple-darwin17.7.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```
To ensure it's not because xcode-select is selecting Command Line Tools:
```
$ xcode-select -p
/Applications/Xcode.app/Contents/Developer
```
Just to be safe, I'll investigate further.

Reply:
If you go back far enough, the Command Line Tools are an add-on to Xcode, not an alternative to it. For those Xcode versions we require both to be installed. For new enough Xcode versions, it's viable to have Xcode installed but not the CLT. We also want to support that case.

Reply:
@jmroot Is it actually viable to have only Xcode without the CLTs? It was my understanding that we always require the CLTs because some ports expect `/usr/include` to contain headers. Is that no longer the case? Do we support only having Xcode installed but not the CLTs?

Reply:
Is something like this better? Assume CLT is in "/" for < 10.9 and not setting DEVELOPER_DIR if CLT not installed. 
```tcl
if {[vercmp $macosx_version 10.9] >= 0} {
    set cltpath "/Library/Developer/CommandLineTools"
} else {
    set cltpath "/"
}
# If CLT not installed, do not set developer_dir.
if {![file isdirectory [file join $cltpath usr include]] || ![file executable [file join $cltpath usr bin make]]} {
    return ""
}
return ${cltpath}
```
Referenced how to check if CLT not installed from _check_xcode_version proc

Reply:
> It was my understanding that we always require the CLTs because some ports expect /usr/include to contain headers. Is that no longer the case? Do we support only having Xcode installed but not the CLTs?

Requiring /usr/include already isn't viable on 10.14. According to jeremyhu the Xcode-only configuration has been working for quite some time already: https://lists.macports.org/pipermail/macports-dev/2017-July/036045.html

Reply:
> Is something like this better? Assume CLT is in "/" for < 10.9 and not setting DEVELOPER_DIR if CLT not installed.

That does look better. However CLT does not install /usr/include on 10.14+, and I'm not sure what effect DEVELOPER_DIR=/ would have on old Xcode versions where the CLT is reliant on having a full Xcode installation as well. Someone needs to test on old OS versions at some point (@kencu?)

Also, we have an existing developer_dir variable with a default value determined by `macports::set_developer_dir`. It seems like this should be integrated with that somehow, or using the same name would be confusing.

Reply:
>Also, we have an existing developer_dir variable with a default value determined by macports::set_developer_dir. It seems like this should be integrated with that somehow, or using the same name would be confusing.

Yeah I guess it's confusing since then we have DEVELOPER_DIR preferring CLT but developer_dir preferring what's selected by `xcode-select`. I don't think it causes any conflicts besides in the [final fallback](https://github.com/macports/macports-base/blob/1d6588d178b251c86abc37cbea5ab580e2b85fd1/src/port1.0/portconfigure.tcl#L1081) of `portconfigure::find_developer_tool` where developer_dir is used.

If we should unify them, we could move the logic of selecting CLT over Xcode to `macports::set_developer_dir`.

If not, we can set DEVELOPER_DIR to developer_dir if CLT not found. But it doesn't actually mattter since if DEVELOPER_DIR isn't set, it looks for `xcode-select` which is what `macports::set_developer_dir` already does.

Reply:
Handled macOS < 10.9 and CLT not found in 074a501